src/algebra/homology/local_cohomology.lean

```diff
@@ -186,4 +194,94 @@ begin
+  functor.initial (ideal_powers_to_self_le_radical J) :=
+{ out := λ J', begin
+  apply @zigzag_is_connected _ _ _,
+  { intros j1 j2,
+    apply relation.refl_trans_gen.single,
+    -- The inclusions `J^n1 ≤ J'` and `J^n2 ≤ J'` always form a triangle, based on
+    -- which exponent is larger.
+    cases le_total (unop j1.left) (unop j2.left) with h,
+      right, exact ⟨costructured_arrow.hom_mk (hom_of_le h).op (of_as_true trivial)⟩,
+      left, exact ⟨costructured_arrow.hom_mk (hom_of_le h).op (of_as_true trivial)⟩ },
+  { obtain ⟨k, hk⟩ := ideal.exists_pow_le_of_le_radical_of_fg _ _ J'.2
+      (is_noetherian_def.mp hR _),
+    refine ⟨costructured_arrow.mk _⟩,
+    exact (op k),
+    exact ⟨⟨hk⟩⟩ },
+  end }
+
+section
+/-
+Running into a universe issue on the next definition:
+using {R : Type u} as above causes an error.
+-/
+
+/-- Local cohomology (defined in terms of powers of `J`) agrees with local
+cohomology computed over all ideals with radical containing `J`. -/
+def iso_self_le_radical
+  {R : Type} [comm_ring R] (J : ideal R) [is_noetherian R R] (i : ℕ) :
+  local_cohomology.of_self_le_radical J i ≅ local_cohomology J i :=
+(local_cohomology.iso_of_final
+  (ideal_powers_to_self_le_radical J) (self_le_radical_diagram J) i).symm
+≪≫ has_colimit.iso_of_nat_iso (iso.refl _)
+
+end
 end local_cohomology_equiv
+
+section
+/-
+TODO: Show that local cohomology depends only on `J.radical`.
+This should essentially follow from `local_cohomology.iso_self_le_radical` above,
+since the full subcategories `self_le_radical J` and `self_le_radical K` are... the same.
+Below is an incomplete attempt.
+-/
+
+variables {R : Type} [comm_ring R]
+
+def self_le_radical.cast {J K : ideal R} (hJK : J.radical = K.radical) :
+self_le_radical J ⥤ self_le_radical K :=
+full_subcategory.map (λ L hL, begin
+                        rw ← ideal.radical_le_radical_iff at ⊢ hL,
+                        exact hJK.symm.trans_le hL,
+                      end)
+
+instance self_le_radical.is_equivalence {J K : ideal R} (hJK : J.radical = K.radical) :
+is_equivalence (self_le_radical.cast hJK) :=
+{ inverse := self_le_radical.cast hJK.symm,
+  unit_iso := by tidy,
+  counit_iso := by tidy }
+
+def of_self_le_radical.iso_of_same_radical  {J K : ideal R} (hJK : J.radical = K.radical) (i : ℕ) :
+  (diagram (self_le_radical.cast hJK ⋙ self_le_radical_diagram K) i) ≅
+  (diagram (self_le_radical_diagram J) i) :=
+diagram_comp (self_le_radical.cast hJK) (self_le_radical_diagram K) i
+
+theorem self_le_radical.iso_of_same_radical [is_noetherian R R] {J K : ideal R}
+  (hJK : J.radical = K.radical)  (i : ℕ) :
+of_self_le_radical J i ≅ of_self_le_radical K i :=
+begin
+  unfold of_self_le_radical,
+  unfold of_diagram,
+  sorry,
+  -- have key :=
+  --   diagram_comp (self_le_radical.cast hJK) (self_le_radical_diagram K) i,
+  -- have key2 :=
+  --   has_colimit.iso_of_equivalence (self_le_radical.cast hJK.symm).as_equivalence,
+end
```

Comment:
This is the main incomplete step here. Partly this is awkward because `local_cohomology.diagram` is a bifunctor. Also, this should be a `def` but is currently a `theorem` to avoid timeouts.